src/js/components/conversation.jsx

```diff
@@ -62,7 +62,32 @@ export class ConversationComponent extends Component {
     }
 
     render() {
-        const messages = this.props.conversation.messages.map((message) => {
+        const processedMessages = [...this.props.conversation.messages];
+        let lastAuthor;
+        processedMessages.forEach((message, index) => {
+            const author = message.role === 'appUser' ? message.role : `${message.role}_${message.name}`;
+
+            if (!lastAuthor) {
+                lastAuthor = author;
+                message.firstInGroup = true;
```

Comment:
I'm not sure if I like mutating the message objects like this. Maybe the algorithm could instead group messages into separate arrays, then use those arrays to power the render function instead of looking at the boolean flags?